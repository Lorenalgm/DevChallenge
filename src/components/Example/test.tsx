import { render, screen } from '@testing-library/react';

import Main from './index';

describe('<Main />', () => {
  it('should render heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /devChallenge/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({
      'background-color': '#0e0f10',
    });
  });
});
