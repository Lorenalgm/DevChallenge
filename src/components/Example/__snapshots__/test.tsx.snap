// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
.c0 {
  background-color: #0e0f10;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

<main
  class="c0"
>
  <img
    alt="Logo"
    class="c1"
    src="/img/logo.png"
  />
  <h1
    class="c2"
  >
    DevChallenge
  </h1>
</main>
`;
